clients/src/main/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManager.java

```diff
@@ -188,12 +184,12 @@ private void onFailedResponse(final long currentTimeMs, final Throwable exceptio
         if (exception == Errors.GROUP_AUTHORIZATION_FAILED.exception()) {
             log.debug("FindCoordinator request failed due to authorization error {}", exception.getMessage());
             KafkaException groupAuthorizationException = GroupAuthorizationException.forGroupId(this.groupId);
-            backgroundEventHandler.add(new ErrorEvent(groupAuthorizationException));
+            fatalError = Optional.of(groupAuthorizationException);
             return;
         }
 
         log.warn("FindCoordinator request failed due to fatal exception", exception);
-        backgroundEventHandler.add(new ErrorEvent(exception));
+        fatalError = Optional.of(exception);
```

Comment:
seems we're never clearing this fatal error and we should right? we could consider clearing it when the request completes, or maybe better when a new request is generated on poll? thoughts?

Reply:
It's cleared in the `AbstractHeartbeatRequestManager` at this point, but I don't know if it should also be cleared elsewhere instead or in addition to `AbstractHeartbeatRequestManager.poll()`.